tests/probes/yamlfilecontent/test_probes_yamlfilecontent_key.xml

```diff
@@ -55,14 +75,24 @@
       <ind-def:state state_ref="oval:0:ste:2"/>
     </ind-def:yamlfilecontent_test>
 
-    <ind-def:yamlfilecontent_test version="1" id="oval:0:tst:3" check="all" comment="error">
+    <ind-def:yamlfilecontent_test version="1" id="oval:0:tst:3" check="all" comment="true">
       <ind-def:object object_ref="oval:0:obj:3"/>
     </ind-def:yamlfilecontent_test>
 
     <ind-def:yamlfilecontent_test version="1" id="oval:0:tst:4" check="all" comment="false">
       <ind-def:object object_ref="oval:0:obj:4"/>
     </ind-def:yamlfilecontent_test>
 
+    <ind-def:yamlfilecontent_test version="1" id="oval:0:tst:5" check="all" comment="true">
+      <ind-def:object object_ref="oval:0:obj:5"/>
+      <ind-def:state state_ref="oval:0:ste:5"/>
+    </ind-def:yamlfilecontent_test>
+
+    <ind-def:yamlfilecontent_test version="1" id="oval:0:tst:6" check="all" comment="true">
+      <ind-def:object object_ref="oval:0:obj:6"/>
+      <ind-def:state state_ref="oval:0:ste:6"/>
+    </ind-def:yamlfilecontent_test>
+
```

Comment:
I would consider adding also a test which doesn't match any map in the yaml file with `comment="false"`.

Reply:
What is the purpose? What should be different from `get_non_existing_scalar` case?

Reply:
Oh sorry, I didn't expressed it correctly, I meant test which doesn't match anything inside a map, for example `.status.conditions[:]['nonexistent']`. I believe it should exercise a different path in the code compared to the `get_non_existing_scalar` case.

Reply:
Done.